packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_single.dart

```diff
@@ -92,56 +86,70 @@ class _RetainedSizeColumn extends ColumnData<HeapClassStatistics> {
         );
 
   @override
-  int getValue(HeapClassStatistics classStats) => classStats.total.retainedSize;
-
-  @override
-  bool get supportsSorting => true;
+  int getValue(SingleClassStats classStats) => classStats.objects.retainedSize;
 
   @override
   bool get numeric => true;
 
   @override
-  String getDisplayValue(HeapClassStatistics classStats) => prettyPrintBytes(
+  String getDisplayValue(SingleClassStats classStats) => prettyPrintBytes(
         getValue(classStats),
         includeUnit: true,
         kbFractionDigits: 1,
       )!;
 }
 
-class SnapshotStatsTable extends StatelessWidget {
-  const SnapshotStatsTable({
+class ClassesTableSingle extends StatefulWidget {
+  const ClassesTableSingle({
     Key? key,
     required this.item,
-    required this.dataId,
     required this.controller,
   }) : super(key: key);
 
-  final SnapshotListItem item;
-
-  final int dataId;
+  final SnapshotInstanceItem item;
 
   final DiffPaneController controller;
 
-  static final _shallowSizeColumn = _ShallowSizeColumn();
+  @override
+  State<ClassesTableSingle> createState() => _ClassesTableSingleState();
+}
 
-  static final List<ColumnData<HeapClassStatistics>> _columns = [
+class _ClassesTableSingleState extends State<ClassesTableSingle>
+    with AutoDisposeMixin {
+  late SingleHeapClasses _classes;
+  final ColumnData<SingleClassStats> _shallowSizeColumn = _ShallowSizeColumn();
+  late final List<ColumnData<SingleClassStats>> _columns =
+      <ColumnData<SingleClassStats>>[
     _ClassNameColumn(),
     _InstanceColumn(),
     _shallowSizeColumn,
     _RetainedSizeColumn(),
   ];
 
+  void _initWidget() {
+    _classes = widget.item.classesToShow() as SingleHeapClasses;
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _initWidget();
+  }
+
+  @override
+  void didUpdateWidget(covariant ClassesTableSingle oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.item != oldWidget.item) _initWidget();
+  }
+
   @override
   Widget build(BuildContext context) {
-    final data = item.statsToShow.classStats;
-    return FlatTable<HeapClassStatistics>(
-      keyFactory: (e) => Key(e.heapClass.fullName),
-      data: data,
-      dataKey: 'snapshot-stats-$dataId',
+    return FlatTable<SingleClassStats>(
       columns: _columns,
-      selectionNotifier: item.selectedClassStats,
-      onItemSelected: (classStats) =>
-          controller.selectedClassName.value = classStats?.heapClass.fullName,
+      data: _classes.classes,
+      dataKey: widget.item.id.toString(),
```

Comment:
add a prefix to this instead of just using the number. Something that uniquely identifies this table, e.g. 'class-stats-${widget.item.id}'. This dataKey needs to be unique all across devtools so having keys '1', '2', etc increases the risk for a collision